LambdaS3FileZipper/Aws/S3FileRetriever.cs

```diff
@@ -11,15 +14,23 @@ public class S3FileRetriever : IFileRetriever
 		private const int MaxConcurrentDownloads = 10;
 
 		private readonly IAwsS3Client s3Client;
+		private readonly ILog log;
 
 		public S3FileRetriever(IAwsS3Client s3Client)
 		{
 			this.s3Client = s3Client;
+
+			this.log = LogProvider.GetCurrentClassLogger();
 		}
 
 		public async Task<string> Retrieve(string bucket, string resource, CancellationToken cancellationToken = default(CancellationToken))
 		{
 			var files = await s3Client.List(bucket, resource, cancellationToken);
+			if (files.Any() == false)
+			{
+				log.Warn("There are no files listed under {bucket}:{resource}; nothing to ZIP", bucket, resource);
+				throw new ArgumentException($"There are no files listed under {bucket}:{resource}; nothing to ZIP");
```

Comment:
🤔 `ArgumentException` or should we throw a custom one?

From [MSDN](https://docs.microsoft.com/en-us/dotnet/api/system.argumentexception):
> The exception that is thrown when one of the arguments provided to a method is not valid.

Reply:
So, technically this is a client/consumer error. I was thinking about returning a specific HTTP status code for example, but I wasn't sure how it would work in the context of AWS Lambda.

I can create a more specific exception type, something like `S3ResourceNotFoundException`.